front/lib/plans/subscription.ts

```diff
@@ -46,62 +46,170 @@ export const internalSubscribeWorkspaceToFreeTestPlan = async ({
   if (!workspace) {
     throw new Error(`Cannot find workspace ${workspaceId}`);
   }
-  // We end the active subscription if any
-  const activeSubscription = await Subscription.findOne({
-    where: { workspaceId: workspace.id, status: "active" },
-  });
+  const now = new Date();
 
-  if (activeSubscription) {
-    await activeSubscription.update({
-      status: "ended",
-      endDate: new Date(),
+  return front_sequelize.transaction(async (t) => {
+    // We end the active subscription if any
+    const activeSubscription = await Subscription.findOne({
+      where: { workspaceId: workspace.id, status: "active" },
+      transaction: t,
     });
-  }
 
-  // We return the default subscription to FREE_TEST_PLAN
-  const freeTestPlan: PlanAttributes = FREE_TEST_PLAN_DATA;
+    if (activeSubscription) {
+      await activeSubscription.update(
+        {
+          status: "ended",
+          endDate: now,
+        },
+        { transaction: t }
+      );
+    }
 
-  return {
-    status: "active",
-    subscriptionId: null,
-    stripeSubscriptionId: null,
-    stripeCustomerId: null,
-    startDate: null,
-    endDate: null,
-    paymentFailingSince: null,
-    plan: {
-      code: freeTestPlan.code,
-      name: freeTestPlan.name,
-      stripeProductId: null,
-      billingType: freeTestPlan.billingType,
-      limits: {
-        assistant: {
-          isSlackBotAllowed: freeTestPlan.isSlackbotAllowed,
-          maxMessages: freeTestPlan.maxMessages,
+    const plan: PlanAttributes = FREE_NO_PLAN_DATA;
+
+    return {
```

Comment:
Let's move this code to an helper since it's shared between here and `Authenticator`.

Reply:
(done)